src/main/java/com/mongodb/spark/sql/connector/read/partitioner/ShardedPartitioner.java

```diff
@@ -54,7 +55,8 @@
  * <p>Uses the chunks collection and partitions the collection based on the sharded collections
  * chunk ranges.
  *
- * <p><strong>Note:</strong> Does not support collections sharded using hashed shard keys.
+ * <p><strong>Note:</strong> Does not support collections sharded using hashed shard keys or
+ * compound shard keys.
```

Comment:
[just a comment]
Indeed, this is something we missed in the original [discussion](https://github.com/mongodb/mongo-spark/pull/63#discussion_r813419181). If [`$lt`](https://www.mongodb.com/docs/manual/reference/operator/query/lt/)&Co were to support comparing a compound "field" against a compound value (which is what the server does when computes chunks and decides which chunk a document goes into, and what `$sort` does, so technically this appear to be possible), then `ShardedPartitioner` and all other partitioners could support compound keys. It is worth noting that other partitioners would, unlike `ShardedPartitioner`, still not be able to support non-unique partition keys.